JSONArray is not a JSONObject

When using QZ Tray 2.1 the follow code fails:

2.0 format:
```js
printData = [
    {
        type: 'raw',
        format: 'image',
        data: 'assets/img/image_sample_bw.png',
        options: { language: 'escp', dotDensity: 'single' }
    },  
    '\x1B' + '\x40',          // init
    '\x1D' + '\x56'  + '\x01' // partial cut (new syntax)
];
```

2.1 format:
```js
printData = [
    {
        type: 'raw',
        format: 'image',
        flavor: 'file',
        data: 'assets/img/image_sample_bw.png',
        options: { language: 'escp', dotDensity: 'single' }
    },
    '\x1B' + '\x40',          // init
    '\x1D' + '\x56'  + '\x01' // partial cut (new syntax)
];
```

```
[ERROR] 2019-10-17 12:43:26,673 @ qz.utils.PrintingUtilities:?
	Failed to print
org.codehaus.jettison.json.JSONException: JSONArray[1] is not a JSONObject.
	at org.codehaus.jettison.json.JSONArray.getJSONObject(JSONArray.java:277)
	at qz.printer.action.PrintImage.parseData(Unknown Source)
	at qz.utils.PrintingUtilities.processPrintRequest(Unknown Source)
	at qz.ws.PrintSocketClient.processMessage(Unknown Source)
	at qz.ws.PrintSocketClient.onMessage(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor10.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.eclipse.jetty.websocket.common.events.annotated.CallableMethod.call(CallableMethod.java:70)
	at org.eclipse.jetty.websocket.common.events.annotated.OptionalSessionCallableMethod.call(OptionalSessionCallableMethod.java:68)
	at org.eclipse.jetty.websocket.common.events.JettyAnnotatedEventDriver$2.run(JettyAnnotatedEventDriver.java:210)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.base/java.lang.Thread.run(Thread.java:834)
[TRACE] 2019-10-17 12:43:26,680 @ qz.utils.PrintingUtilities:?
	Returning processor back to pool
```

--- a/src/qz/printer/PrintOptions.java
+++ b/src/qz/printer/PrintOptions.java
@@ -231,7 +231,10 @@ public class PrintOptions {
         }
 
         //grab any useful service defaults
-        PrinterResolution defaultRes = output.getNativePrinter().getResolution().value();
+        PrinterResolution defaultRes = null;
+        if (output.isSetService()) {
+            defaultRes = output.getNativePrinter().getResolution().value();
+        }
         if (defaultRes != null) {
             //convert dphi to unit-dependant density ourselves (to keep as double type)
             defOptions.density = (double)defaultRes.getFeedResolution(1) / psOptions.getUnits().getDPIUnits();

--- a/src/qz/utils/PrintingUtilities.java
+++ b/src/qz/utils/PrintingUtilities.java
@@ -15,7 +15,8 @@ import qz.printer.action.ProcessorFactory;
 import qz.ws.PrintSocketClient;
 
 import java.awt.print.PrinterAbortException;
-import java.util.*;
+import java.util.Arrays;
+import java.util.Locale;
 
 public class PrintingUtilities {
 
@@ -45,8 +46,16 @@ public class PrintingUtilities {
         //grab first data object to determine type for entire set
         JSONObject data = printData.optJSONObject(0);
 
-        Format format;
+        //Check for RAW type to coerce COMMAND format handling
+        Type type;
         if (data == null) {
+            type = Type.RAW;
+        } else {
+            type = Type.valueOf(data.optString("type", "RAW").toUpperCase(Locale.ENGLISH));
+        }
+
+        Format format;
+        if (type == Type.RAW) {
             format = Format.COMMAND;
         } else {
             format = Format.valueOf(data.optString("format", "COMMAND").toUpperCase(Locale.ENGLISH));

